experimental/rsocket/RSocketRequestHandler.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/io/async/EventBase.h>
+
+// TODO migrate yarpl/v/Flowable to yarpl/Flowable
+#include "yarpl/v/Flowable.h"
+
+#include "src/ReactiveStreamsCompat.h"
+#include "src/StandardReactiveSocket.h"
+
+#include "src/Payload.h"
+#include "src/StandardReactiveSocket.h"
+#include "src/StreamState.h"
+
+namespace rsocket {
+
+/**
+ * RequestHandler APIs to handle requests on an RSocket connection.
+ *
+ * This is most commonly used by an RSocketServer, but due to the symmetric
+ * nature of RSocket, this can be used on the client as well.
+ */
+class RSocketRequestHandler {
+ public:
```

Comment:
I am happy to see how this interface doesn't have to do anything with setup/resume initialization.